File: benchmark/nodejs/ack.js
function ack(m, n) {
    if (m === 0) {
        return n + 1;
    }
    if (n === 0) {
        return ack(m - 1, 1);
    }
    return ack(m - 1, ack(m, n - 1));
}

const start = Date.now();
const result = ack(3, 6);
const end = Date.now();
console.log(`Result: ${result}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/src/ruby/ack.rb
def ack(m, n)
  return n + 1 if m == 0
  return ack(m - 1, 1) if n == 0
  ack(m - 1, ack(m, n - 1))
end

start = Time.now.to_f * 1000
result = ack(3, 6)
finish = Time.now.to_f * 1000
puts "Result: #{result}"
puts "Time: #{(finish - start).round}ms"

